### PR TITLE
docs(lockFileMaintenance): update list of supported lockfiles

### LINKED ISSUE #19292: Update list of supported lock files for `lockFileMaintenance` config option
### Describe the proposed change(s).

Can one of the maintainers check if the list of supported lock files is correct?

Is there a way to automate the list generation? I would expect our code to "know" what lockfiles it can handle. Maybe we can export that into the docs somehow?

Quote from docs: [^docs]

> ## lockFileMaintenance
> 
> This feature can be used to refresh lock files and keep them up-to-date.
> "Maintaining" a lock file means recreating it so that every dependency version within it is updated to the latest.
> Supported lock files are:
> 
> - `package-lock.json`
> - `yarn.lock`
> - `pnpm-lock.yaml`
> - `composer.lock`
> - `Gemfile.lock`
> - `poetry.lock`
> - `Cargo.lock`
> - `jsonnetfile.lock.json`
> 
> Others may be added via feature request.

[^docs]: https://docs.renovatebot.com/configuration-options/#lockfilemaintenance

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Update list of supported lockfiles
- Order list from A-Z

## Context

Closes #19292

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -1412,14 +1412,21 @@ This feature can be used to refresh lock files and keep them up-to-date.
 "Maintaining" a lock file means recreating it so that every dependency version within it is updated to the latest.
 Supported lock files are:
 
-- `package-lock.json`
-- `yarn.lock`
-- `pnpm-lock.yaml`
+- `.terraform.lock.hcl`
+- `Cargo.lock`
 - `composer.lock`
+- `flake.lock` // Nix lockfile?
```
Comment (JamieMagee) [inline docs/usage/configuration-options.md:1418]: Yeah, this is a nix lockfile

Comment (rarkins) [inline docs/usage/configuration-options.md:1418]:
```suggestion
- `flake.lock`
```

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -1412,14 +1412,21 @@ This feature can be used to refresh lock files and keep them up-to-date.
 "Maintaining" a lock file means recreating it so that every dependency version within it is updated to the latest.
 Supported lock files are:
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1413]: Updating the list manually has some drawbacks:

- I can't know all possible lockfiles and their matching managers
- For me, finding the lockfiles in the package manager official docs can be hard
- The programmers that write the manager code know the supported lockfiles
- putting new entries in the list is likely to mess up the alphabetical ordering by accident

Can we automate the list?

How about this:

1. Each package manager has a new constant which has an array with strings of valid lock files for that manager, and the name of the package manager
2. Docs generating code grabs alls the consts, and creates one A-Z sorted lists of supported lockfiles formats plus the package manager name

So the final format:

```markdown
List of supported lockfiles:

- `lockfile format` (manager name)
- `lockfile format 1`, `lockfile format 2` (for manager that supports multiple lockfiles)
- and so on...
```

**Review by HonkingGoose — COMMENTED** [review]
Idea to automate the list, let me know if this idea deserves a new issue?

File: docs/usage/configuration-options.md (hunk 2 above)
Comment (viceice) [reply docs/usage/configuration-options.md:1413]: sounds good, please open an issue to track

Comment (HonkingGoose) [reply docs/usage/configuration-options.md:1413]: Here's the tracking issue:

- #19340

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -1412,14 +1412,21 @@ This feature can be used to refresh lock files and keep them up-to-date.
...
-- `poetry.lock`
-- `Cargo.lock`
+- `gradle.lockfile`
 - `jsonnetfile.lock.json`
+- `package-lock.json`
+- `Pipfile.lock`
+- `pnpm-lock.yaml`
+- `poetry.lock`
+- `pyproject.toml`
+- `requirements.lock` // Helm V3 lockfile?
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1427]:
```suggestion
- `Chart.lock`
- `Chart.yaml`
```

Are these the right lockfiles for Helm v3?

Do we support Helm V2 (the older version)? If yes, maybe we should keep the `requirements.yaml` and `requirements.lock` in the list of supported lockfiles? 

Quote from a Helm v3 preview blog: [^blog]

> The Chart dependency management system moved from `requirements.yaml` and `requirements.lock` to `Chart.yaml` and `Chart.lock`, meaning that charts that relied on the helm dependency command will need some tweaking to work in Helm 3.

[^blog]: https://helm.sh/blog/helm-3-preview-pt5/#helm

Comment (rarkins) [reply docs/usage/configuration-options.md:1427]: Sounds like just `Chart.lock`

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -1412,14 +1412,21 @@ This feature can be used to refresh lock files and keep them up-to-date.
...
+- `requirements.txt` // pipenv lockfile?
```
[the remark this answers is not in the document]
Comment (rarkins) [reply docs/usage/configuration-options.md:1429]: Kind of. it's more like TS->JS than package file -> lock file. i.e. `requirements.txt` is a package file itself, but `pip-compile` can generate it from `requirements.in`

**Comment by viceice** [comment]
conflicted 😉

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -1413,15 +1413,22 @@ This feature can be used to refresh lock files and keep them up-to-date.
...
+- `Cargo.lock`
+- `Chart.lock`
 - `composer.lock`
+- `flake.lock`
 - `Gemfile.lock`
-- `poetry.lock`
-- `Cargo.lock`
+- `gradle.lockfile`
 - `jsonnetfile.lock.json`
+- `package-lock.json`
```
Comment (viceice) [inline docs/usage/configuration-options.md:1424]:
```suggestion
- `package-lock.json`
- `packages.lock.json`
```
nuget / dotnet lockfile 😉

**Review by viceice — APPROVED** [review]
🤷‍♂️

**Pull request merged**